Mage/src/main/java/mage/abilities/costs/common/PayEnergyCost.java

```diff
@@ -30,14 +30,14 @@ public PayEnergyCost(PayEnergyCost cost) {
     @Override
     public boolean canPay(Ability ability, Ability source, UUID controllerId, Game game) {
         Player player = game.getPlayer(controllerId);
-        return player != null && player.getCounters().getCount(CounterType.ENERGY) >= amount;
+        return player != null && player.getCountersCount(CounterType.ENERGY) >= amount;
     }
 
     @Override
     public boolean pay(Ability ability, Game game, Ability source, UUID controllerId, boolean noMana, Cost costToPay) {
         Player player = game.getPlayer(controllerId);
-        if (player != null && player.getCounters().getCount(CounterType.ENERGY) >= amount) {
-            player.getCounters().removeCounter(CounterType.ENERGY, amount);
+        if (player != null && player.getCountersCount(CounterType.ENERGY) >= amount) {
+            player.removeCounters(CounterType.ENERGY.getName(), amount, source, game);
```

Comment:
@JayDi85 So this is the error of not firing event on removing counters.
Preventing direct access to Counters is to prevent this to happen without noticing.

Reply:
Only through the Player's methods should you be able to add/remove counters. This is not overcomplicated. This is safer through not exposing methods that are unsafe.

Reply:
hmm, I see the problem

Reply:
Looks like a wrong usage in cards too:

![shot_240527_231719](https://github.com/magefree/mage/assets/8344157/bd92ca8a-badd-41a2-a6da-ae678fbadec8)

Reply:
More
![shot_240527_231947](https://github.com/magefree/mage/assets/8344157/31352a56-6644-4d05-91ae-117e40a4f1e1)

Reply:
1. I think `Counters.addCounter` can be renamed to `withCounter` due usage logic (works with new counters).

![shot_240527_232553](https://github.com/magefree/mage/assets/8344157/525db398-d9d3-48a3-ba3e-d6c67b82e36f)
![shot_240527_232220](https://github.com/magefree/mage/assets/8344157/9abade00-8a8f-4ceb-a2d0-91e3c98b71cc)

2. `removeAllCounters` can be deleted and replaced by one by one remove to raise correct events. Maybe it will require new card/player methods to support good events lifecycle same as `Player.removeCounters`

Reply:
1. Actually I removed that method and made the test calling it use the other addCounter method instead.

2. Done. I went through all of https://scryfall.com/search?q=o%3A%22remove+all%22 and https://scryfall.com/search?q=o%3A%2Floses%3F+all%5B%5E%2C.%5D*counter%2F&unique=cards&as=grid&order=name to make sure to simplify the cards to use the new `removeAllCounters` methods in both `Player` & `Card`. It is a 50 files change though, so if you want I can split that in a followup PR.

Of note, I still plan to prevent direct access to `CardImpl::counters`, but that one is more complex (and there is already Tawnoss Coffin as weird card that will need its own ad hoc solution)